### PR TITLE
Parallel module

### PR DESCRIPTION
**Pull request by dhrone**
* Added new ``parallel`` module to ``luma.core.interface``
* Moved ``parallel`` class from ``luma.core.interface.serial`` to ``luma.core.interface.parallel`` and renamed it to ``bitbang_6800``
* Updated documentation
* Updated CHANGES.rst

### CODE REVIEW

File: CHANGES.rst [hunk 1]
```diff
@@ -4,6 +4,9 @@ ChangeLog
 +------------+---------------------------------------------------------------------+------------+
 | Version    | Description                                                         | Date       |
 +============+=====================================================================+============+
+| **1.16.2** | * Added new parallel interface module                               | 2020/09/20 |
```
Comment (thijstriemstra) [inline CHANGES.rst:7]: replace release date with TBA, it'll be updated day it's going on pypi

File: luma/core/interface/parallel.py [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+    :param RS: The GPIO pin register select (RS) line (low for command, high
+        for data)
+    :type RS: int
+    :param E: The GPIO pin to connect the enable (E) line to.
+    :type E: int
+    :param PINS: The GPIO pins that form the data bus (a list of 4 or 8 pins
+        depending upon implementation ordered from LSD to MSD)
+    :type PINS: list[int]
+
+    .. versionadded:: 1.15.1
```
Comment (thijstriemstra) [inline luma/core/interface/parallel.py:40]: i would update the version nr and use latest

File: luma/core/interface/parallel.py [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+Encapsulates sending commands and data over a parallel-bus interface.
+"""
+
+from time import sleep
+from luma.core import lib
+
+
+__all__ = ["bitbang_6800"]
+
+#: Default amount of time to wait for a pulse to complete if the device the interface is connected to requires a pin to be 'pulsed' from low to high to low for it to accept data or a command
```
Comment (thijstriemstra) [inline luma/core/interface/parallel.py:15]: put this comment on multiple lines.

File: luma/core/interface/parallel.py [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+    :type E: int
+    :param PINS: The GPIO pins that form the data bus (a list of 4 or 8 pins
+        depending upon implementation ordered from LSD to MSD)
+    :type PINS: list[int]
+
+    .. versionadded:: 1.15.1
+    """
+
+    def __init__(self, gpio=None, pulse_time=PULSE_TIME, **kwargs):
+
```
Comment (thijstriemstra) [inline luma/core/interface/parallel.py:44]: remove newline

File: luma/core/interface/serial.py [hunk 5]
```diff
@@ -667,8 +572,8 @@ class pcf8574(i2c):
        connected to P7, and the RS value to indicate command is low, your
        initialization would look something like...
```
Comment (thijstriemstra) [inline luma/core/interface/serial.py:573]: like: instead of like...

**Review by thijstriemstra — CHANGES_REQUESTED** [review]

File: luma/core/interface/parallel.py (hunk 2 above)
Comment (thijstriemstra) [reply luma/core/interface/parallel.py:40]: or leave it as is and add a version marker in the module docstring with latest

File: CHANGES.rst (hunk 1 above)
Comment (dhrone) [reply CHANGES.rst]: Changed as suggested.  Will do the same with ``luma.lcd`` and ``luma.oled``

File: luma/core/interface/parallel.py (hunk 2 above)
Comment (dhrone) [reply luma/core/interface/parallel.py]: Changed to 1.16.2

File: luma/core/interface/parallel.py (hunk 3 above)
Comment (dhrone) [reply luma/core/interface/parallel.py]: Changed as suggested

File: luma/core/interface/parallel.py (hunk 4 above)
Comment (dhrone) [reply luma/core/interface/parallel.py]: Removed

File: luma/core/interface/serial.py (hunk 5 above)
Comment (dhrone) [reply luma/core/interface/serial.py]: Changed as suggested

**Review by rm-hull — APPROVED** [review]
Happy for this to be merged once @thijstriemstra approves. I will bump the version and release out to pypi. Do we think 1.16.2 or 1.17.0?

**Review by thijstriemstra — APPROVED** [review]
1.16.2 sounds good to me @rm-hull

**Comment by rm-hull** [comment]
luma.core 1.16.2 released to pypi: https://pypi.org/project/luma.core/1.16.2/

**Pull request merged**